libraries/adsk/adsklib/colorcorrect.mtlx

```diff
@@ -1,116 +1,119 @@
-      <parameter name="channels" type="string" value="rgb" />
-    </swizzle>
-    <swizzle name="outputAlpha" type="float">
-      <input name="in" type="color4" nodename="if_unpremultiply_condition" />
-      <parameter name="channels" type="string" value="a" />
-    </swizzle>
-    <swizzle name="extractColorForHsv" type="color3">
-      <input name="in" type="color4" nodename="if_premultiply_condition" />
-      <parameter name="channels" type="string" value="rgb" />
-    </swizzle>
-    <swizzle name="extractAlphaForGain" type="float">
-      <input name="in" type="color4" nodename="if_premultiply_condition" />
-      <parameter name="channels" type="string" value="a" />
-    </swizzle>
-    <constant name="inputColor" type="color3">
-      <parameter name="value" type="color3" value="0.5, 0.5, 0.5" interfacename="color" />
-    </constant>
-    <constant name="inputAlpha" type="float">
-      <parameter name="value" type="float" value="1" interfacename="alpha" />
-    </constant>
-    <combine2 name="combineInput" type="color4">
-      <input name="in1" type="color3" nodename="inputColor" />
-      <input name="in2" type="float" nodename="inputAlpha" />
-    </combine2>
-    <output name="outColor" type="color3" nodename="outputColor" />
-    <output name="outAlpha" type="float" nodename="outputAlpha" />
-  </nodegraph>
-  
+
+   <!-- Sample color correct interface. Versioned and namespaced -->
+   <nodedef name="ND_colorcorrect" node="colorcorrect" nodegroup="adjustment" version="1.0" isdefaultversion="true" namespace="adsk">
+      <parameter name="color" type="color3" value="0.5, 0.5, 0.5" />
+      <parameter name="alpha" type="float" value="1" />
+      <input name="HSV" type="vector3" value="0, 1, 1" />
+      <input name="color_gain" type="color3" value="1, 1, 1" />
+      <input name="alpha_gain" type="float" value="1" />
+      <input name="color_offset" type="color3" value="0, 0, 0" />
+      <input name="alpha_offset" type="float" value="0" />
+      <input name="color_gamma" type="color3" value="1 1, 1" />
+      <input name="alpha_gamma" type="float" value="1" />
+      <input name="unpremultiply_input" type="boolean" value="false" />
+      <input name="premultiply_input" type="boolean" value="false" />
+      <parameter name="clamp_color" type="boolean" value="false" />
+      <input name="min_color" type="color3" value="0, 0, 0" />
+      <input name="max_color" type="color3" value="1, 1, 1" />
+      <parameter name="clamp_alpha" type="boolean" value="false" />
+      <input name="min_alpha" type="float" value="0" />
+      <input name="max_alpha" type="float" value="1" />
+      <output name="outColor" type="color3" />
+      <output name="outAlpha" type="float" />
+   </nodedef>
+
+   <!-- Sample color correct implementation. Versioned and namespaced -->
+   <nodegraph name="NG_colorcorrect" nodedef="adsk:ND_colorcorrect" namespace="adsk" version="1.0">
```

Comment:
"version" attribute is added back in for nodedef<->nodegraph matching logic.
Done in a few places.